### PR TITLE
create activity log

### PR DESCRIPTION
**Pull request by phonghenry14**

### CODE REVIEW

File: app/controllers/admin/teams_controller.rb [hunk 1]
```diff
@@ -49,7 +50,7 @@ def destroy
   end
 
   private
-  def team_params 
+  def team_params
```
Comment (khanhkhanh) [inline app/controllers/admin/teams_controller.rb]: Need a space before []

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class Admin::UsersController < ApplicationController
   before_action :authenticate_user!
   before_action :admin_user
+  before_action :create_log, only: [:create, :update, :destroy]
 
   def index
     @users = User.normal.paginate page: params[:page]
```
Comment (khanhkhanh) [inline app/controllers/admin/users_controller.rb]: paginate without per_page?

File: app/controllers/admin/teams_controller.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 class Admin::TeamsController < ApplicationController
   before_action :authenticate_user!
   before_action :admin_user
+  before_action :create_log, only: [:create, :update, :destroy]
 
   def index
     @teams = Team.paginate page: params[:page]
```
Comment (khanhkhanh) [inline app/controllers/admin/teams_controller.rb]: Need per_page here :)

File: app/controllers/admin/teams_controller.rb [hunk 4]
```diff
@@ -32,10 +33,10 @@ def edit
   end
 
   def update
-    @team = Team.find params[:id]    
+    @team = Team.find params[:id]
```
Comment (vjnhat) [inline app/controllers/admin/teams_controller.rb]: duplicode, refactor it .

File: app/controllers/admin/users_controller.rb [hunk 5]
```diff
@@ -1,9 +1,10 @@
 class Admin::UsersController < ApplicationController
   before_action :authenticate_user!
   before_action :admin_user
```
Comment (vjnhat) [inline app/controllers/admin/users_controller.rb]: try before_action :authenticate_user!, :admin_user
if it run normal, i think it is better

File: app/controllers/admin/teams_controller.rb (hunk 4 above)
Comment (phonghenry14) [inline app/controllers/admin/teams_controller.rb]: Cậu xem lại hộ tớ với, duplicode đoạn trên hay đoạn dưới

File: app/models/activity.rb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+class Activity < ActiveRecord::Base
+  enum action: [:create1, :update1, :destroy1]
```
Comment (dieunb) [inline app/models/activity.rb]: bad naming

File: app/models/concerns/activity_logs.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+module ActivityLogs
+  def create_activity_log action_name, class_name
+    @activity = Activity.new
+    @activity.action = action_name
+    @activity.object = class_name
+    @activity.description = @activity.action + " " + @activity.object
```
Comment (dieunb) [inline app/models/concerns/activity_logs.rb]:
```
"#{@activity.action} #{@activity.object}"
```

File: app/models/user.rb [hunk 8]
```diff
@@ -18,7 +18,24 @@ class User < ActiveRecord::Base
...
+  after_update :log_update
+  after_destroy :log_destroy
+
   def is_admin?
     role == Settings.user.role.admin
   end
+
+  private
+  def log_create
+    create_activity_log "create1", self.class.name
```
Comment (dieunb) [inline app/models/user.rb]: Store in Setting file

File: app/models/user.rb [hunk 9]
```diff
@@ -18,7 +18,24 @@ class User < ActiveRecord::Base
...
+  end
+
+  def log_update
+    create_activity_log "update1", self.class.name
```
Comment (dieunb) [inline app/models/user.rb]: Store in Setting file

File: app/models/user.rb [hunk 10]
```diff
@@ -18,7 +18,24 @@ class User < ActiveRecord::Base
...
+  end
+
+  def log_destroy
+    create_activity_log "destroy1", self.class.name
```
Comment (dieunb) [inline app/models/user.rb]: Store in Setting file

File: app/controllers/admin/skills_controller.rb [hunk 11]
```diff
@@ -1,4 +1,7 @@
 class Admin::SkillsController < ApplicationController
+  before_action :authenticate_user!
```
Comment (vjnhat) [inline app/controllers/admin/skills_controller.rb]: try before_action :authenticate_user!, :admin_user
if it run normal, it will be better

**Pull request merged**